### PR TITLE
Fixed PyRight issues.

### PR DESCRIPTION
**Pull request by DragonMoffon**
The latest version of PyRight is a little over zealous.. There where two flagged non-issues. One where it didn't realise that the code would escape early, and another which is all pyglets fault.

### CODE REVIEW

**Review by cspotcode — COMMENTED** [review]
When `# type: ignore` is due to pyglet issues, we have a convention of adding a comment linking to the pyglet issue.  Should we keep that convention?  You can search for `# pending` to see it.

For example: https://github.com/pythonarcade/arcade/blob/development/arcade/sound.py#L143

File: arcade/physics_engines.py [hunk 1]
```diff
@@ -53,7 +54,7 @@ def _circular_check(player: Sprite, walls: List[SpriteList]):
         vary *= 2
 
 
-def _move_sprite(moving_sprite: Sprite, walls: List[SpriteList], ramp_up: bool) -> List[Sprite]:
+def _move_sprite(moving_sprite: Sprite, walls: List[SpriteList], ramp_up: bool) -> List[Union[Sprite, BasicSprite]]:
```
Comment (cspotcode) [inline arcade/physics_engines.py]:
```suggestion
def _move_sprite(moving_sprite: Sprite, walls: List[SpriteList[SpriteType]], ramp_up: bool) -> List[SpriteType]:
```

I think this should be generic.  The returned list contains wall sprites, so it'll return whatever type of BasicSprite (sub)class is contained in the `walls` SpriteLists.

File: arcade/gui/widgets/text.py [hunk 2]
```diff
@@ -488,8 +487,10 @@ def __init__(
 
         # Set how fast the mouse scroll wheel will scroll text in the pane.
         # Measured in pixels per 'click'
-        self.scroll_speed = scroll_speed if scroll_speed is not None \
-                            else font_size
+        self.scroll_speed = (
+            scroll_speed if scroll_speed is not None
+            else font_size
+        )
```
Comment (cspotcode) [inline arcade/gui/widgets/text.py:493]: Is this related to pyright?  It's fine if not, the code looks better when parens.  Just wondering.

Comment (DragonMoffon) [reply arcade/gui/widgets/text.py:493]: The line was too long and had originally used a slash which I find hard to read. If you have a recommendation on how to format it I'm happy to update it

Comment (cspotcode) [reply arcade/gui/widgets/text.py:493]: No recommendation, I wanted to confirm it wasn't pyright choking on backslashes or something.  I agree with your formatting.

File: arcade/physics_engines.py (hunk 1 above)
Comment (pushfoo) [reply arcade/physics_engines.py]: @DragonMoffon Bumping since this PR fixes issues which flag #1876 as broken.
